internal/database/inmemory.go

```diff
@@ -225,26 +215,23 @@ func CalculateAverage(samples []uplink.GPUStatSample) uplink.GPUStatSample {
 		sumMaxGraphicsClock += sample.MaxGraphicsClock
 		sumMemoryClock += sample.MemoryClock
 		sumMaxMemoryClock += sample.MaxMemoryClock
-
 		if sample.Time < minTime {
 			minTime = sample.Time
 		}
-
+		// Addressing potential duplicate PID handling
 		for _, proc := range sample.RunningProcesses {
-			// TODO: What if multiple samples have a process with the same PID?
-			processesMap[proc.Pid] = proc
+			processesMap[proc.Pid] = proc // Overwrites if duplicate, ensuring only the latest is kept
```

Comment:
NIT: I'm not sure i fully agree this is the right semantics, could we keep the TODO?